hist/hist/src/Haxis.cxx

```diff
@@ -216,6 +265,71 @@ void TH1::SetAxisRange(Axis_t xmin, Axis_t xmax, Option_t *axis)
    theAxis->SetRange(bin1, bin2);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Set the histogram bar offset along X and Y axis.
+///
+/// axis specifies which axis ("x","y"), default = "xy"
+///
+///  - if axis="xy" `offset` is set along X and Y axis
+///  - if axis="x"  `offset` is set along the X axis and 0. along the Y axis
+///  - if axis="y"  `offset` is set along the Y axis and 0. along the X axis
+
+void TH1::SetBarOffset(Float_t offset, Option_t *axis)
+{
+   TString opt = axis;
+   opt.ToLower();
+
+   Bool_t barox = kFALSE; if (opt.Contains("x")) barox = kTRUE;
+   Bool_t baroy = kFALSE; if (opt.Contains("y")) baroy = kTRUE;
+
+   if (barox && baroy) {
+      fBarOffset = Short_t(1000*offset);
+      return;
+   }
+   if (barox) {
+      fBarOffset = Short_t(1000*offset) + 1000;
+      return;
+   }
+   if (baroy) {
+      fBarOffset = Short_t(1000*offset) + 2000;
+      return;
+   }
+
+   Warning("SetBarOffset","Invalid option");
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Set the histogram bar width along X and Y axis.
+///
+/// axis specifies which axis ("x","y"), default = "xy"
+///
+///  - if axis="xy" `width` is set along X and Y axis
+///  - if axis="x"  `width` is set along the X axis and 1. along the Y axis
+///  - if axis="y"  `width` is set along the Y axis and 1. along the X axis
+
+void TH1::SetBarWidth(Float_t width, Option_t *axis)
+{
+   TString opt = axis;
+   opt.ToLower();
+
+   Bool_t barwx = kFALSE; if (opt.Contains("x")) barwx = kTRUE;
+   Bool_t barwy = kFALSE; if (opt.Contains("y")) barwy = kTRUE;
+
```

Comment:
Here one should check if width inside [0..1] range.
Also if none x and y are specified, one should enable both like:

    if (!barwx && !barwy) barwx = barwy = kTRUE;

Reply:
I agree with the range checking. 
I do not with setting both in case of invalid option. The default is "xy" (see the include file). 
      if one set an invalid option the setter will do nothing and print a warning.